core/templates/rid_owner.h

```diff
@@ -266,7 +271,7 @@ class RID_Alloc : public RID_AllocBase {
 		uint32_t idx_element = idx % elements_in_chunk;
 
 		uint32_t validator = uint32_t(id >> 32);
-		if (unlikely(validator_chunks[idx_chunk][idx_element] & 0x80000000)) {
+		if (unlikely(validator >= 0x7FFFFFFF || validator_chunks[idx_chunk][idx_element] & 0x80000000)) {
```

Comment:
`>= 0x7FFFFFFF` is the same as `& 0x80000000` I think (in release it will probably check the bit rather than do a comparison, as I'm assuming this is cheaper). Not that using the former is bad, just noting this is the same thing.

Reply:
Ah true, but I'm still not sure this is required, same thing as the other cases, it compares the `validator_chunks` with the `validator` so this should be caught in the existing code.

Reply:
True this case checks against the full validator so will catch, thank you